Color Gradients for Storybook

superhawk610 opened the pull request:
**Related Issue:** #130 
**Related PR:** #136 

**Summary:**
While Storybook is generally used for displaying full Components, I've used it in personal projects as a style guide for newcomers. With this in mind, I think having an easy overview of all available colors in Guppy's palette is a great tool for new contributors and experienced users alike.

I took a page from @j-f1's book and deployed this to Netlify for easy demoing - link [here](https://happy-sinoussi-976836.netlify.com/).

**Screenshots/GIFs:**
![storybook-colors](https://user-images.githubusercontent.com/18172185/44297040-5899c000-a298-11e8-9175-bbee365c9339.gif)

joshwcomeau commented on src/utils.js:
```diff
@@ -240,3 +242,138 @@ export const safeEscapeString = (val: string) =>
+  }
+
+  throw new Error(`Unrecognized color code: '${colorCode}'`);
+};
```
Guppy has an NPM dependency, [color](https://www.npmjs.com/package/color), that should be able to do a lot of this parsing. We'd still need methods for luma and `contrastingColor`, but we should be able to defer to it for a lot of this stuff :)

superhawk610 commented:
Alright, I added a second story with the entire Material Design palette. I also tweaked the `utils` logic to use the existing `color` package, and added a short disclaimer at the top of each story explaining its purpose.

I also added in a small tweak in `manager-head.html` to remove the second scrollbar displayed on the right side of the preview panel (I use Chrome Canary so I'm not sure if this is a common problem, if it's not just let me know and I'll drop that from the PR.)

The Netflify preview has also been updated - visit it [here](https://happy-sinoussi-976836.netlify.com/).

superhawk610 commented:
Here's a preview of the new story:

![storybook-colors](https://user-images.githubusercontent.com/18172185/44303747-d9979c80-a316-11e8-9973-e0c245dd835a.gif)

joshwcomeau commented on .storybook/colors.stories.js:
```diff
@@ -0,0 +1,163 @@
+import Heading from '../src/components/Heading';
+import { contrastingColor } from '../src/utils';
+import { COLORS as guppyColors } from '../src/constants';
+import { default as materialColors } from 'material-colors/dist/colors';
```
Ahh, I might have been unclear earlier D:

What I meant was that we should quickly grep through the project to find the 10ish colors we're likely already using, as a subset of all the colors in `COLORS`.

I have mixed feelings about what you've done. On the one hand it seems convenient to have additional colors available, although I wonder if we want to encourage people to use additional colors (I think we likely have more than enough already, even for edge-case things);

Sorry for not being clearer :( FWIW I really like how you handled the `info` distinction and the ColorSwatch components and such. I think we can probably use exactly the same structure, just with different input colors?

I can also be the one who scans through and finds the "common" ones if you prefer! I don't wanna trouble you any more with this PR haha.

superhawk610 replied:
No problem 😁 I did think it was a little strange to include the entire Material palette, I definitely think keeping things focused makes more sense. Now I see what you were saying, there are definitely quite a few colors included that only appear once or not at all.

I'll update this tomorrow according to those specs and then you can review the common colors and make sure I got everything you had in mind!

joshwcomeau replied:
Yay, thanks for understanding! Sounds good.

(I just noticed I ended one of my paragraphs in my comment in a semicolon, hah.)

joshwcomeau commented on .storybook/manager-head.html:
```diff
@@ -0,0 +1,10 @@
+  * By default, Storybook has the preview panel set to `overflow: auto`.
+  * For whatever reason, this causes two scrollbars to show up for all
+  * content. This removes the outer scrollbar.
+  */
```
Interesting! I had no idea there was an HTML file exposed. Good troubleshooting :)

joshwcomeau commented on src/utils.js:
```diff
@@ -240,3 +242,79 @@ export const safeEscapeString = (val: string) =>
+    console.warn(e + ', defaulting to #fff');
+    return '#fff';
+  }
+};
```
Looks great!

joshwcomeau commented:
Closing this, as it's been replaced by #152
